### ISSUE TITLE
Can't create pages/routes/blocks/menu in demo

### ISSUE THREAD

**Issue by tristanbes** [Sylius/Sylius#3139]
Hi,

I didn't manage to create either routes, either pages, or blocks.
For example, when I try to create a simple block, I got an error message on a hidden field:

![capture d ecran 2015-08-17 a 09 54 43](https://cloud.githubusercontent.com/assets/346010/9300425/17544ad0-44c6-11e5-8d26-ee86529d9a48.png)

**edit**: It's the same when trying creating a menu

**Comment by pjedrzejewski** [comment]
Hi Tristan! Thanks for reporting that. I will try to reproduce this issue and get back to you soon.

**Comment by psyray** [comment]
Maybe another collateral damage of PHPCR
 #2696
https://github.com/doctrine/DoctrinePHPCRBundle/pull/193

**Comment by psyray** [comment]
Like in the above issue, maybe @dbu, @smith77 or @dantleech can help
I don't investigate but I'm sure this was caused by a recent Doctrine PHPCR change

**Comment by dbu** [comment]
can it be that js is broken? did you not select a parent? or is the mapping of the block erronous?

**Comment by tristanbes** [comment]
The error is triggered on a hidden field and it seems that there is a value (`/cms/blocks`)

**Comment by dbu** [comment]
did you run the initializers? is there a path /cms/blocks in the phpcr repository? `app/console doctrine:phpcr:node:dump /cms/blocks`

@psyray is the parent field hidden by a sylius admin class or something directly from the CmfBlockBundle?

**Comment by tristanbes** [comment]
This is out of my juridiction as I tryed only in the demo mode,  i'll let @psyray take it over :-)

**Comment by dbu** [comment]
@tristanbes maybe just try to run the console command to see if there is a setup issue? also try `app/console doctrine:phpcr:node:dump /cms --depth=2` if you have no /cms/blocks...

**Comment by psyray** [comment]
For now, no problem for me with sylius while not updating vendors.
Error "This value is not valid" is not reported and saving is working fine.
So error is not within Sylius code change but maybe doctrinePHPCRBundle code change.
Will investigate more and update vendors to see if error is reported.

**Comment by psyray** [comment]
@dbu 
The parent field is the parentDocument of the CmfBlockBundle defined in the block-bundle here :
https://github.com/symfony-cmf/BlockBundle/blob/master/Model/AbstractBlock.php#L48

This field is used in all of the Sylius ContentBundle forms to define the parent of a document to prevent user to type in and mistake.
Maybe there is a better way to handle this ?
The CMF Admin Bundle of Sonata works like this, so I decide to do the same way and force parent in a hidden or read-olny field (only difference is that sonata use tree to select parent)
![selection_069](https://cloud.githubusercontent.com/assets/1230954/9412289/29a59f1e-482b-11e5-8f44-2f41c6c7c45b.png)


The Sylius demo seems to works now.
I think something (cache or something) was not correctly initialized, so the error occurs. 
Maybe @pjedrzejewski could tell us more.

**Comment by psyray** [comment]
To close this issue, I confirm that after a vendors update everything is working properly.
So this bug can be closed ;)

Error came from a CMF repository not initialized.

@dbu Thanks for help
@tristanbes Thanks for report

**Comment by psyray** [comment]
@pjedrzejewski 
If you want, you could assign the CMF bugs to me ;)
If I could I wili check and debug ASAP

**Comment by dbu** [comment]
great, thanks a lot!

if anybody has an idea how to provide a better error message, i am happy to review a pull request :-) the problem basically is when the form transformers do not find the parent node, they just say its invalid. one thing to do could also be a command that checks for existence of the necessary root nodes (but that would be the same as having a command that checks if the doctrine schema has been create - just running the `doctrine:phpcr:repository:init` again would also solve the problem.)

**Comment by jpiazza** [comment]
I started to test Sylius yesterday and I have many issues until I installed sylius:v0.15.0 and not the sylius-standar version.
For this issue and the version I have installed I just add in table phpcr_nodes the '/cms/routes' with depth 2 and works. 
Hope this be useful for anyone.
Greetings from Argentina.

**Comment by informatico-madrid** [comment]
Hi. Same problem here. doctrine:phpcr:repository:init dont create /cms/routes. Im in update process from sylius 0.13 to 0.15 anyone know why of this?
thankx

**Comment by psyray** [comment]
Hi,

What's the output of the `doctrine:phpcr:repository:init` command ?
Did you run the all the migration script after update ?
If not, use the `doctrine:migrations:migrate` command to update db to latest version.
And then launch `doctrine:schema:update --force` to update the remaining schema.

**Comment by pamil** [comment]
It may be related to symfony-cmf/RoutingBundle#251.

**Comment by informatico-madrid** [comment]
I have not a production environment. I only update composer.json and related files in app/config. and update my src code extended when it is necessary.
doctrine:database:drop
doctrine:database:create
doctrine:schema:create
cache:clear
octrine:phpcr:repository:init
sylius:install:setup

**Comment by psyray** [comment]
did you update the vendors correctly ?
`php composer.phar update --prefer-dist --profile -vvv`

**Comment by informatico-madrid** [comment]
I had not done, but it makes no difference

**Comment by psyray** [comment]
I have tried from scratch, no problem for me everything is working fine.

You should not run these commands :
```
doctrine:database:drop
doctrine:database:create
doctrine:schema:create
cache:clear
doctrine:phpcr:repository:init
```
but this one
`sylius:install:database`
I recommend using this command because it will initialize everything, not only db but also rbac, phpcr, sample data ...

Result is
```
$ php app/console sylius:install:database
Creating Sylius database for environment dev.
It appears that your database already exists. Would you like to reset it? (y/N) y
 8/8 [||||||||||||||||||||||||||||] 100%
Warning! This will erase your database. Your current environment is dev.
Load sample data? (y/N) y
Loading sample data...
 3/3 [||||||||||||||||||||||||||||] 100%
```
And after that you must run
```
php app/console sylius:install:setup
```

Result is
```
$ php app/console sylius:install:setup
Please enter list of locale codes, separated by commas or just hit ENTER to use "en_US". For example "en_US, de_DE".
In which language your customers can browse the store?  
Adding anglais (États-Unis).
Please enter list of currency codes, separated by commas or just hit ENTER to use "USD". For example "USD, EUR, GBP".
In which currency your customers can buy goods?  
Adding dollar des États-Unis.
Please enter list of country codes, separated by commas or just hit ENTER to use "US". For example "US, PL, DE".
To which countries you are going to sell your goods?  
Adding États-Unis.
Please enter a list of channels, separated by commas or just hit ENTER to use "DEFAULT". For example "WEB-UK, WEB-DE, MOBILE".
On which channels are you going to sell your goods?  
Adding DEFAULT.
Create your administrator account.
Your firstname: john
Lastname: doe
E-Mail: john.doe@sylius.org
Choose password: 
Repeat password: 
```

**Comment by psyray** [comment]
> I had not done, but it makes no difference

If you update composer.json, you **MUST** launch 
` php composer.phar update --prefer-dist --profile -vvv`

**Comment by informatico-madrid** [comment]
Thankx psyray.
Excuse me. My bad English has created confusion. I did not mean that "update --prefer-dist" was not essential. I meant to run "" did not solve the problem.
 
I cant run "sylius:install:database" this command give me error: 
'''
[OutOfBoundsException]                                   
Reference to: (Sylius.Taxonomy.Category) does not exist 
'''

the error is generated by "sylius:install:sample-data"
"sylius:install:sample-data" runs in first "doctrine:fixtures:load"
output "doctrine:fixtures:load":
'''
Careful, database will be purged. Do you want to continue Y/N ?y
  > purging database
  > loading [10] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadPaymentMethodsData
  > loading [10] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadGroupsData
  > loading [10] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadLocalesData
  > loading [10] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadImagesData
  > loading [10] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadTaxonomiesData
  > loading [10] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadCountriesData
  > loading [10] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadContactTopicData
  > loading [10] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadRbacData
  > loading [10] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadCurrencyData
  > loading [20] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadPromotionsData
  > loading [20] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadProductOptionData
  > loading [20] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadApiData
  > loading [20] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadProductAttributeData
  > loading [20] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadContactRequestData
  > loading [20] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadZonesData
  > loading [20] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadUsersData
  > loading [30] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadTaxationData
  > loading [30] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadProductArchetypeData
  > loading [30] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadShippingData
  > loading [40] Sylius\Bundle\FixturesBundle\DataFixtures\ORM\LoadChannelData                                           
  [OutOfBoundsException]                                   
  Reference to: (Sylius.Taxonomy.Category) does not exist 
''' 
I dont know if this are related with de main problem, remember phpcr route: '/cms/routes are missing'
Meaby yes. 
But i dont want load fixtures!!
in this point i have a question: 
"doctrine:fixtures:load" or "doctrine:phpcr:fixtures:load" create node: "/cms/routes" ?

I resove yet the problem create my own command this command run sylius:install:setup and create this missing node. now see like are working good.

**Comment by psyray** [comment]
I think you have 2 problems :
- First one: As you understand, if you don't install the fixtures the /cms/routes doesn't create, because of this PR https://github.com/symfony-cmf/RoutingBundle/pull/251, you could resolve the route creation by applying the fix #3668  made by @pamil 
- Second one : I think your bundle is not in sync with the Sylius 0.15, did you update correctly all the src/YourBundle files ? Did you have an override of the taxonomy model that can cause the Reference error ?

**Comment by informatico-madrid** [comment]
i dont finish yet check my bundle is in sync with SYlius 0.15, im working on that now. First i will to finish. and if problem persist open new issue about reference error. but i dont override taxonomy model.

**Comment by psyray** [comment]
For my own project I use [meld] (http://meldmerge.org/) to check difference between two folders/files

**Comment by informatico-madrid** [comment]
Hey very thankx, i dont know this tools. i will try now ;)

**Comment by sydney-akhann** [comment]
Hi,

Same here, I was not able to create routes because of the initialisation issue with symfony-cmf/RoutingBundle, even after updating the bundle.

I had to install SonataDoctrinePHPCRAdminBundle to make the command doctrine:phpcr:repository:init initialize CmfRoutingBundle

Since SonataDoctrinePHPCRAdminBundle solve the problem maybe it should be added to sylius composer/kernel

**Comment by pamil** [comment]
@sydney-akhann it will be fixed after symfony-cmf/RoutingBundle#318 is merged.

**Comment by dbu** [comment]
that is merged now, but you would need to require
symfony-cmf/routing-bundle 1.4.*@dev for now. we are still in the
process of wrapping up PHP 7 and Symfony 3 support before we can tag a
stable version.

**Comment by pjedrzejewski** [comment]
I am fine with requiring dev version for now so that it works.

**Comment by bosnianmostar** [comment]
Hey guys, I did 

 doctrine:phpcr:repository:init 

and looks like I get new issues, 
acctualy now I can't see my store because of this error:

"An exception has been thrown during the rendering of a template ("Doctrine is not available.") in SyliusWebBundle:Frontend:layout.html.twig at line 97."

Could anyone help me?

**Comment by psyray** [comment]
Seems it's a [general Sylius bug](http://stackoverflow.com/questions/36404596/sylius-error-after-composer-update-on-sylius-0-17), please open a new issue

**Comment by psyray** [comment]
An issue has been opened 5 days ago #4716 
Please search issue before posting
